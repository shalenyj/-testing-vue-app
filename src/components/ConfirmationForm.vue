<template>
  <TForm name="confirm-form" @submit="handleConfirm" submit-title="confirm">
    <div class="input-wrap">
      <label for="name">Name: </label>
      <input
        id="name"
        v-model="name"
        type="text"
        required
        placeholder="Tell us your name"
      />
    </div>
  </TForm>
</template>

<script>
import TForm from "@/elements/TForm";

export default {
  name: "ConfirmationForm",
  components: {
    TForm,
  },
  data() {
    return {
      name: null,
    };
  },
  methods: {
    handleConfirm() {
      this.$emit('submit', this.name)
    },
  },
};
</script>
