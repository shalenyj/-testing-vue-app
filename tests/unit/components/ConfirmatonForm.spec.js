import { mount } from "@vue/test-utils";

import ConfirmationForm from "@/components/ConfirmationForm";

describe("@/elements/ConfirmationForm", () => {
  it("should match snapshot", () => {
    const wrapper = mount(ConfirmationForm);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should verify if elements exist", () => {
    const wrapper = mount(ConfirmationForm);
    expect(wrapper.find(".buttons").exists()).toBeTruthy();
    expect(wrapper.find(".t-button").exists()).toBeTruthy();
    expect(wrapper.find("form").exists()).toBeTruthy();
  });
});
