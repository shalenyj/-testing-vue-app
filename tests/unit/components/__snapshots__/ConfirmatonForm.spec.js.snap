// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/elements/ConfirmationForm should match snapshot 1`] = `
<form
  name="name"
>
  <div
    class="input-wrap"
  >
    <label
      for="name"
    >
      Name: 
    </label>
     
    <input
      id="name"
      placeholder="Tell us your name"
      required="required"
      type="text"
    />
  </div>
   
  <section
    class="buttons"
  >
    <button
      class="t-button"
      type="submit"
    >
      
  confirm

    </button>
  </section>
</form>
`;
